core/core-impl/src/main/java/io/openshift/appdev/missioncontrol/tracking/segment/SegmentAnalyticsProvider.java

```diff
@@ -0,0 +1,91 @@
+package io.openshift.appdev.missioncontrol.tracking.segment;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.logging.Logger;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.Resource;
+import javax.enterprise.concurrent.ManagedExecutorService;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Produces;
+
+import com.segment.analytics.Analytics;
+import com.segment.analytics.messages.MessageBuilder;
+import com.segment.analytics.messages.TrackMessage;
+
+import io.openshift.appdev.missioncontrol.core.api.LaunchEvent;
+
+/**
+ * Provides the {@link Analytics} instance to be used
+ *
+ * @author <a href="mailto:alr@redhat.com">Andrew Lee Rubinger</a>
+ * @author <a href="mailto:tschotan@redhat.com">Tako Schotanus</a>
+ */
+@ApplicationScoped
+public class SegmentAnalyticsProvider {
+
+    private static final Logger log = Logger.getLogger(SegmentAnalyticsProvider.class.getName());
+
+    private static final String NAME_EVENT_LAUNCH = "launch";
+    private static final String KEY_OPENSHIFT_PROJECT_NAME = "openshiftProjectName";
+    private static final String KEY_GITHUB_REPO = "githubRepo";
+
+    //TODO Externalize into sysprop
+    private static final String KEY = "oOlNiAf3K5MDwpd4ErD2ZPRe6z3Ckk7w";
+
+    private Analytics analytics;
+
+    @Resource
+    private ManagedExecutorService async;
+
+    @PostConstruct
+    public void initAnalytics() {
+        // TODO Externalize as above
+        final String key = KEY;
+        analytics = Analytics.builder(key).networkExecutor(async).build();
+        log.finest(() -> "Using Segment analytics with key: " + key);
+	}
+
+    public void onEvent(@Observes LaunchEvent launch) throws IOException {
+        createTrackingTask(
+                launch.getUser(),
+                launch.getId(),
+                launch.getGithubRepo(),
+                launch.getOpenshiftProjectName());
+	}
+
+    public void createTrackingTask(final String userId,
```

Comment:
That happens when you takes someone else's code and adjust it. The name doesn't make sense either.